### DISCUSSION TITLE
How to determine minimum version of setuptools

### DISCUSSION THREAD

**Discussion by penguinpee** [pypa/setuptools#3618] [Q&A]
I'm trying to determine the minimum version of setuptools required for my package. Is there a tool or script for that?

Background:

When I moved to a pure pyproject.toml setup (no setup.cfg or setup.py), I failed to set the minimum version to >=61. I only discovered that oversight when my package failed to build in the Fedora build environment. In f36 the shipped version of setuptools is 59.6.0.

I now lowered the setuptools requirement by moving the project table from [pyproject.toml](https://gist.github.com/penguinpee/e32be4fb77aba77dec560ecfd827e903#file-pyproject-toml) into [setup.cfg](https://gist.github.com/penguinpee/e32be4fb77aba77dec560ecfd827e903#file-setup-cfg) allowing it to build for f36. The package builds with setuptools <=56. Yet, I want to know what minimum version is required for my setup, so I can specify it correctly this time around. It's currently set to >=42, which it was before a [fellow Pythonian pointed out](https://github.com/penguinpee/PyMunin3/pull/3) to me that it needed to be at least 61.

**Comment by webknjaz** [comment]
I usually read the changelog at https://setuptools.pypa.io/en/latest/history.html, sometimes I do some testing to verify the findings and in rare cases grep the source, check git blame use issue/PR search.

Sometimes, there's a note about the version in place where a certain feature is documented. If it's not there, feel free to contribute a docs PR adding the `.. versionadded::` Sphinx RST directive there.

**Reply by penguinpee** [reply]
Thank you @webknjaz. I was actually hoping for something a little more helpful, a little easier to use. I'm familiar with the History document you pointed to[^1].

Following your advice, I would need to scan all the _Breaking Changes_ sections until I hit something that is in my configuration (daunting) or keep building my project, lowering the setuptools version until it breaks (doable/scriptable).

[^1]: I have not so sweet memories of endlessly scrolling through it after yet another link to it, saying the information is there. Finding it is another matter.

**Reply by penguinpee** [reply]
I just tried, having to start somewhere, with `requires = ["setuptools>=42", "setuptools_scm[toml]>=6.2"]`. That gave me:

```python
    The user requested setuptools<42
    setuptools-scm[toml] 7.0.5 depends on setuptools
    setuptools-scm[toml] 7.0.5 depends on setuptools>=42; extra == "toml"
```

So, the baseline is >=42, which I had before. Since that will pull in the highest available version when using pip, it doesn't help much. In the build environments of Fedora, and I assume other distros, the highest available version is not determined by what pip can find, but by what's available in the repos, which I learned the hard way as explained in my initial question.
